### PR TITLE
[WIP] archive symlinks

### PR DESCRIPTION
**Pull request by mwhooker**
There seems to be an alternate PR at #51. I'm not sure where either of these are at, but this one looks complete. Anything you remember needing to be done besides real-world testing, @cbednarski ?

### CODE REVIEW

**Comment by cbednarski** [comment]
@mwhooker I think all I did was add a failing test but did not actually fix the bug.

**Comment by cbednarski** [comment]
@mwhooker I'd say pull this in, then pull in #59 and see if the tests pass. I did not end up digging into why this is broken. Maybe #59 fixes something else.

File: archive/archive_test.go [hunk 1]
```diff
@@ -242,6 +242,28 @@ func TestArchive_dirWithSymlink(t *testing.T) {
...
+	path := filepath.Join(testFixture("archive-symlink-file"), "link")
+	r, err := CreateArchive(path, new(ArchiveOpts))
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	expected := []string{
+		"link/",
+		"link/foo.txt",
+	}
```
Comment (waniji) [inline archive/archive_test.go]: Missing test files that added with b4f681e9b65b887d5a3f85adedbb2ac253af837c commit.

```diff
 expected := []string{
-        "link/",
-        "link/foo.txt",
+        "deeper/",
+        "deeper/adeeperlink",
+        "deeper/linklink",
+        "deeper/linklinklink",
+        "link",
 }
```

**Comment by waniji** [comment]
Here, we get the symlink target rather than the symlink itself.
https://github.com/hashicorp/atlas-go/blob/b4ce6286d134f590ab1cd5554f6d346cf92e24a2/archive/archive.go#L353-L371

However, since does not `return nil`, it also acquires the symbolic link itself, and an error will occur.

```diff
     if err := copyConcreteEntry(tarW, subpath, target, info); err != nil {
    	return err
     }

     if info.IsDir() {
             return filepath.Walk(target, copyDirWalkFn(
                     tarW, target, subpath, opts, vcsInclude))
     }
+    return nil
 }

 return copyConcreteEntry(tarW, subpath, path, info)
```

**Comment by waniji** [comment]
This symbolic link path is incorrect.
https://github.com/hashicorp/atlas-go/blob/b4ce6286d134f590ab1cd5554f6d346cf92e24a2/archive/test-fixtures/archive-symlink/link/link

```diff
-test-fixtures/archive-symlink/real/
+../real/
```

**Comment by waniji** [comment]
I made some comments.
I hope this will be of some help :smile:

**Comment by SwampDragons** [comment]
incorporated @waniji's excellent comments; tests are passing.

**Comment by waniji** [comment]
thank you!

**Pull request merged**